src/get.rs

```diff
@@ -45,13 +45,18 @@ impl Default for Options {
     }
 }
 
-/// Setup a QUIC connection to the provided address.
-async fn setup(opts: Options) -> Result<quinn::Connection> {
+/// Create a quinn client endpoint
+pub fn make_client_endpoint(
+    peer_id: Option<PeerId>,
+    alpn_protocols: Vec<Vec<u8>>,
+    keylog: bool,
+    ipv6: bool,
```

Comment:
nitpicking/bikeshedding: I just noticed you made this `pub` which means I care a little more about the API.  And one boolean argument in a signature makes me squint.  Two makes me hesitate.

One alternative would be to keep accepting the SocketAddr to be connected to and only use it to determine the address family.  Could be a tad confusing though.

Another would be to introduce a custom `enum AddressFamily{IpV4, IpV6}`.

Another would be basically going the builder way is some form, e.g. `IrohEndpointBuilder::new().with_peer_id(peer_id).with_alpn(vec!["a", "b"]).with_keylog(false).with_ipv4().build()`

Reply:
I will just pass in the bind addr then.